02-peer-protocol.md

```diff
@@ -70,6 +70,9 @@ If this fails at any stage, or a node decides that the channel terms
 offered by the other node are not suitable, the channel establishment
 fails.
 
+Note that multiple channels can operate in parallel, but only after
```

Comment:
> introduces a state in the peer

State in the peer already exists? 

Good point about concurrent channel funding, that somehow slipped my mind. Here's a suggestion (pretty much was `lnd` does today): 
   * Allow concurrent channel funding. 
   * Each node has a policy which dictates the max allowed number of in-flight channel funding workflows. If a peer attempts to initiate a new funding workflow that puts the total number of active in-flight funding workflows over this limit, then that attempt is rejected. 
   * Don't trigger retransmission semantics before the `funding_signed` message is sent. This saves nodes from having to persist pending channel state until they've actually committed funds. 
   * Re-introduce temp channel ID's. Dictate a loose schema governing how they're generated: the connecting peer starts from `1` and increments with each new channel, and listening peer starts from `1 << 63` and also increments with each channel. 

The above retains using `utxo`'s for de-multiplexing, allows nodes to only allow a single in-flight funding workflow (if they wish), solves issues with concurrent channel fundings (there's essentially two funding "streams"), and makes "temp chanID's" more predictable (along with avoiding the possibility of collisions).